pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -361,17 +361,42 @@ func (ctrl *PersistentVolumeController) syncUnboundClaim(ctx context.Context, cl
 				return nil
 			}
 
+			// The claim can't be bound to any existing volume. let's check if it can be
+			// dynamically provisioned. If can't, we will emit an event to inform the user
+			// the reason why the claim is pending. otherwise, we will provision a volume
+			// for the claim in an asynchronous goroutine and return immediately.
 			switch {
 			case delayBinding && !storagehelpers.IsDelayBindingProvisioning(claim):
+				// If the claim is in delay binding mode and not annotated with
+				// "volume.kubernetes.io/selected-node" by the scheduler's VolumeBinding
+				// plugin, it means that no active pod is observed by the scheduler to
+				// be using this claim. In this case, we should emit an event to inform
+				// the user that the claim is waiting for a pod to be scheduled.
+				//
+				// If the annotation is present, it means that the scheduler has already
+				// assumed a node for the pod and the claim is in delay binding provisioning
+				// mode. In this case, the branch won't be hit.
 				if err = ctrl.emitEventForUnboundDelayBindingClaim(claim); err != nil {
 					return err
 				}
 			case storagehelpers.GetPersistentVolumeClaimClass(claim) != "":
+				// If the claim has a storage class set, but no available volumes are found,
+				// try to provision a volume. The static provisioning and unset storage class
+				// case, the branch won't be hit.
```

Comment:
"_If the claim has a storage class set, but no available volumes are found,_" again repeats the condition + `// No PV could be found` comment somewhere above.

"_The static provisioning and unset storage class case, the branch won't be hit_" Don't describe what's not hit in this branch - do you really want to list _all_ the cases?

Reply:
Not really, the original purpose was that I didn't understand the reason why this case returned directly, but other cases did not. After looking at implement of this function again, I understood, so I added comments here to make it easier for others to understand it. 

Now, other redundant comments are removed.